test/test_pty.rs

```diff
@@ -0,0 +1,36 @@
+use nix::pty::openpty;
+use nix::unistd::{read, write, close};
+
+#[test]
+fn test_openpty() {
+    // TODO: figure out the right termios settings to pass in here
```

Comment:
Huh, I'm pretty sure it is on Linux-based platforms at least. From my understanding, the terminal emulator spawns a pty, gives the slave (or master? one of them) to the shell program, and draws stuff to the screen based on what it receives on its end. Anyways, that fix did it, both tests passing on all important test cases